client/src/main/java/org/apache/rocketmq/client/producer/DefaultMQProducer.java

```diff
@@ -896,7 +894,6 @@ public QueryResult queryMessage(String topic, String key, int maxNum, long begin
     public MessageExt viewMessage(String topic,
         String msgId) throws RemotingException, MQBrokerException, InterruptedException, MQClientException {
         try {
-            MessageId oldMsgId = MessageDecoder.decodeMessageId(msgId);
             return this.viewMessage(msgId);
         } catch (Exception e) {
         }
```

Comment:
You can rename the 'Exception e' to 'Exception ignored'

Reply:
Good idea

Reply:
Would it be better to print the exception message?

Reply:
Oh, this method is 'Deprecated', just ignore it

Reply:
Ok,I will solve it